packages/task/src/browser/task-schema-updater.ts

```diff
@@ -67,6 +70,41 @@ export class TaskSchemaUpdater {
         }
     }
 
+    /**
+     * Adds given task schema to `taskConfigurationSchema` as `oneOf` subschema.
+     * Replaces existed subschema by given schema if the objects are equal or if the corrresponding `$id` properties are equal.
+     * @param schema subschema for adding to `taskConfigurationSchema`
+     */
+    addSubschema(schema: IJSONSchema): void {
+        const schemaId = schema.$id;
+        const arg = schemaId ? schemaId : schema;
+
+        this.removeSubschema(arg);
+
+        customSchemas.push(schema);
+        this.update();
+    }
+
+    /**
+     * Removes task subschema from `taskConfigurationSchema`.
+     *
+     * @param arg `$id` property of subschema or subschema for removing
+     */
+    removeSubschema(arg: string | IJSONSchema): void {
+        let index = -1;
+        if (typeof arg === 'string') {
+            index = customSchemas.findIndex(existed => !!existed.$id && existed.$id === arg);
+        } else {
+            index = customSchemas.findIndex(existed => existed === arg);
```

Comment:
would `===` always work properly since `arg` is an object in this else-block?

Reply:
I don't want to use something like `deepEqual` here, so I relied on `$id` property or case when references point on the same object.
Updated - now the method removes subschema by `$id` property only.

Reply:
so `$id` is required in the schema, if the user of this function wants both `add` and `remove` operations work properly.